examples/assembly/Add_Array_(AArch64).asm

```diff
@@ -0,0 +1,191 @@
+# This file demonstrates adding 2 arrays using serial, SVE and NEON instructions.
+# Compile with aarch64-linux-gnu-gcc -mcpu=generic+sve (or -march=armv8.2-a+sve),
+# the compiler driver sets up code to call main.
+
+# All functions have the following prototype:
+# void addarrays_{serial,neon,sve} (int *restrict res, int *A, int *B, long N)
+# x0 -> res
+# x1 -> A
+# x2 -> B
+# x3 -> N
+
+.globl addarrays_serial
+.type addarrays_serial, %function
+
+addarrays_serial:
+    mov x4, xzr
+    b .cond
+.loop_body:
+    ldr w5, [x1, x4, lsl 2]
```

Comment:
`# Each int is 4 bytes so we use lsl 2, which means << 2, which is the same as x 4.